docs/content/guide/i18n.ngdoc

```diff
@@ -142,84 +144,179 @@ displaying the date with a timezone specified by the developer.
+later.)
 
-You can read more about the ICU MessageFormat syntax at
-[Formatting Messages | ICU User Guide](http://userguide.icu-project.org/formatparse/messages#TOC-MessageFormat).
 
-This extended syntax is provided by way of the
-`ngMessageFormat` module that your application can depend
-upon (shipped separately as `angular-messageFormat.min.js` and
-`angular-messageFormat.js`.)  A current limitation of the
-`ngMessageFormat` module, is that it does not support
-redefining the `$interpolate` start and end symbols. Only the
-default `{{` and `}}` are allowed.
-
-This syntax extension, while based on MessageFormat, has
-been designed to be backwards compatible with existing
-AngularJS interpolation expressions.  The key rule is simply
-this: **All interpolations are done inside double curlies.**
-The top level comma operator after an expression inside the
-double curlies causes MessageFormat extensions to be
-recognized.  Such a top level comma is otherwise illegal in
-an Angular expression and is used by MessageFormat to
-specify the function (such as plural/select) and it's
-related syntax.
-
-To understand the extension, take a look at the ICU
-MessageFormat syntax as specified by the ICU documentation.
-Anywhere in that MessageFormat that you have regular message
-text and you want to substitute an expression, just put it
-in double curlies instead of single curlies that
-MessageFormat dictates.  This has a huge advantage.  **You
-are no longer limited to simple identifiers for
-substitutions**.  Because you are using double curlies, you
-can stick in any arbitrary interpolation syntax there,
-including nesting more MessageFormat expressions!  Some
-examples will make this clear.  In the following example, I
-will only be showing you the AngularJS syntax.
+#### Messages
 
+Messages immediately follow a selection keyword and are optionally preceded by whitespace.  They are
+written in single curly braces (`{}`).  They may contain Angular interpolation syntax inside them.
+In addition, the `#` symbol is a placeholder for the actual numeric value of the expression.
 
 ### Simple plural example
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have # new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have # new messages}
 }}
 ```
 
-While I won't be teaching you MessageFormat here, you will
-note that the `#` symbol works as expected.  You could have
-also written it as:
+Because these messages can themselves contain Angular expressions, you could also write this as
+follows:
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have {{numMessages}} new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have {{numMessages}} new messages}
```

Comment:
I should actually add a section on best practices but that would touch on many things so I have not done so.

The example you have written is actually wrong.  I mean—it's correct for English, but it's highly unlikely to work if you use regular translation tools.  You're basically embedding a fragment of English text inside a subexpression and expecting that it will work.  The Angular expression typically would become an opaque placeholder to the translator that they can't change (and can't see that it contains "no" inside it.)  Also, in a different language, the position of "no" might need to move around.  This scheme will definitely work if one is only supporting a single language.

The following is also bad practice as it relies on the same type of "concatenation" assumption.  (The best practice is to let the translator see the messages in their entirety and not piecemeal.)

```text
You have {{numMessages, plural, =0 {no} =1 {one} other {#} }} new messages
```